### PR TITLE
Config loader patch

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR makes config loaders patchable by updating methods which use config loaders to support custom config loader support.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by angrybayblade — COMMENTED** [review]
So I have patched every method which uses the config loaders and might have to load the service config in some way or another so we can overwrite commands as we need on the autonomy package to support service config loading.

**Review by dvilelaf — APPROVED** [review]

File: aea/cli/push_all.py [hunk 1]
```diff
@@ -39,7 +39,18 @@
 @registry_flag()
 def push_all(packages_dir: Optional[Path], registry: str) -> None:
     """Push all available packages to a registry."""
+    try:
+        push_all_packages(registry, packages_dir)
+    except ValueError as e:
+        raise click.ClickException(str(e)) from e
```
Comment (DavidMinarsch) [inline aea/cli/push_all.py:45]: I think it would be better to catch the value error earlier, e.g. wrap lines 72-73? Everything else is handled afaict

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]
Where are the tests with some mock config ?

**Comment by angrybayblade** [comment]
> Where are the tests with some mock config ?

We already have the config loader patch tests: https://github.com/valory-xyz/open-aea/blob/0b8d75f5c7a449d8b281936919c086dd28f69962/tests/test_configurations/test_loader.py#L109

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**